game/civ5map.ksy

```diff
@@ -156,8 +151,19 @@ types:
             type: u1
           - id: feature2_type_id
             type: u1
-          - id: unknown
+          - id: resource_amount
             type: u1
+        types:
+          river:
+            seq:
+              - id: river_head
```

Comment:
1. should the prefix be removed?
2. should it be documented?

Reply:
1. By prefix do you mean river_head? It has to be there so we can read the next three bits of river data. It's uncertain what it represents other than the fact that it will be zeroes if there's no river.
2. The documentation for river_head is two lines down in commit https://github.com/kaitai-io/kaitai_struct_formats/pull/243/commits/9922b598b322aa49b390d4da6df019db1718b553

Reply:
1. By prefix do you mean river_head? It has to be there so we can read the next three bits of river data. It's uncertain what it represents other than the fact that it will be zeroes if there's no river.
2. The documentation for river_head is two lines down in commit https://github.com/kaitai-io/kaitai_struct_formats/pull/243/commits/9922b598b322aa49b390d4da6df019db1718b553

Reply:
1 By prefix I mean `river_` part of `id`s. You already call the struct `river`, calling its fields `river` additional seems redundant.
2. By`it` I mean the whole structure.  For example it is unclear where are other derections. They can be possihle encoded in what you call `head`. Could you create a some damn simple map and then manipulate `head` bit-by-bit and load it to the game and observe the effect in order to check this hypothesis?

Reply:
1. Fair point. Names changed to latest names in https://github.com/fechan/kaitai_struct_formats/commit/261040364a2a4e2c94047f3c4156bf7f9c09253c
2. The tiles are tessellating hexagons, so defining only those three edges will define rivers for all valid river spots. This behavior is now documented in the ksy file with a doc definition for the river type as suggested.
3. `river_head` is now `unknown`.